src/olympia/reviewers/views.py

```diff
@@ -1329,7 +1329,7 @@ def get_addon_object(self):
         if not hasattr(self, 'addon_object'):
             # We only need translations on the add-on, no other transforms.
             self.addon_object = get_object_or_404(
-                Addon.objects.get_queryset().only_translations(),
+                Addon.unfiltered.all().only_translations(),
```

Comment:
With that change and the one below we *really* want https://github.com/mozilla/addons-server/issues/14607 cause developers should never be able to see deleted add-ons. So to be safe, maybe implement #14607 first.